var/spack/repos/builtin/packages/dlib/package.py

```diff
@@ -81,3 +88,14 @@ def cmake_args(self):
             )
 
         return args
+
+    @run_after("build")
+    def build_python(self):
+        if self.spec.satisfies("+python"):
+            pip = self.spec["python"].command
+            pip.add_default_arg("-m", "pip")
+            env = EnvironmentModifications()
+            env.set("CMAKE_BUILD_PARALLEL_LEVEL", str(make_jobs))
+            pip.add_default_envmod(env)
+            args = PythonPipBuilder.std_args(self) + [f"--prefix={self.prefix}", "."]
```

Comment:
Briefly looking at the `dlib` setup.py, how does this prevent a second CMake build? Should this not pass the spack build directory somehow?

Reply:
It actually requires a second cmake build as their cmake configuration is currently written.  The python bindings have a second top level cmakelists.txt one of which is driven by setup.py.  There's been a PR open for 3+ years in dlib to fix this and modernize cmake but its stalled out.